modules/wizard/view-wizard-step-home-page.php

```diff
@@ -5,54 +5,60 @@
  * @package Polylang
  *
  * @since 2.7
+ *
+ * @var PLL_Admin_Model $model   `PLL_Admin_Model` instance.
+ * @var array           $options List of Polylang options.
  */
 
-if ( ! defined( 'ABSPATH' ) ) {
-	exit; // Don't access directly.
-}
+defined( 'ABSPATH' ) || exit;
 
-$languages = $this->model->get_languages_list();
-$default_language = count( $languages ) > 0 ? $this->options['default_lang'] : null;
+$languages = $model->languages->get_list();
+$default_language = $model->languages->get_default();
 $home_page_id = get_option( 'page_on_front' );
-$translations = $this->model->post->get_translations( $home_page_id );
+$home_page_id = is_numeric( $home_page_id ) ? (int) $home_page_id : 0;
+$translations = $model->post->get_translations( $home_page_id );
 $untranslated_languages = array();
 $home_page = $home_page_id > 0 ? get_post( $home_page_id ) : null;
-$home_page_language = $this->model->post->get_language( $home_page_id );
+$home_page_language = $model->post->get_language( $home_page_id );
+
+if ( empty( $home_page ) ) {
+	return;
+}
 
 foreach ( $languages as $language ) {
-	if ( ! $this->model->post->get( $home_page_id, $language ) ) {
+	if ( ! $model->post->get( $home_page_id, $language ) ) {
 		$untranslated_languages[] = $language;
 	}
 }
```

Comment:
Part of the code here should be in `display_step_home_page()` as several variables are not used in the html part.

Reply:
Done in 77e9faafdec597c021e8846ecbde2dd9cb6e21a7